onDirectionChange should be optional

zvikarp opened the pull request:
Hi, great plugin!

The parameter 'onDirectionChange' is not required, therefore we should not relay on it. But in line 39 we do: `(_) => widget.onDirectionChange!(isRTL(widget.text)));`

In this PR I fix that issue :)

zvikarp committed:
onDirectionChange should be optional

Mohamed-Emara77 commented:
Yup it took me half an hour to figure it out
however great plugin anyway

zvikarp commented:
Hey, I just came back here after facing this issue again and noticed I had already created a pr to fix it already 🙈 . Is there a reason not to merge this pr? maybe I can help with it.

zvikarp committed:
if text is empty use default direction

zvikarp committed:
code cleanup

MuhammedRefaat commented:
@saveKenny so when we expect to have this update release as a new package version?

saveKenny commented:
I don't have write-access. @AhmedAlYousif can you please merge those changes?

zvikarp commented:
btw, I changed some more code, if the text is an empty string, it would use the default direction of the app. This changes the behavior of the package (although this is the correct way to do it :) ).

For example: if the direction of the app is RTL, and we use this package to wrap a text field, while the text field is empty, the direction is set to LTR. I fixed it and it now stays RTL.

```
bool isRTL(String text) {
    if (text.isEmpty) return Directionality.of(context) == TextDirection.rtl;
    return intl.Bidi.detectRtlDirectionality(text);
  }
```

AhmedAlYousif commented:
Sorry for being late... and thank you @zvikarp for your pull request
